src/game.ml

```diff
@@ -54,13 +99,191 @@ module type SoldierLogic = sig
+          && net_y = 1 && net_x = -1
+          && board.(curr_x - 1).(curr_y + 1) != None
+        then true (* Diagonal up-left -- Yes en passant *)
+        else if
+          basic_valid_square && left_en_passant_able && net_y = 1
+          && net_x = -1
+          && board.(curr_x - 1).(curr_y + 1) = None
+        then true (* Diagonal up-right -- no en passant *)
+        else if
+          basic_valid_square
+          && (not right_en_passant_able)
+          && net_y = 1 && net_x = 1
+          && board.(curr_x + 1).(curr_y + 1) != None
+        then true (* Diagonal up-right -- Yes en passant *)
+        else if
+          basic_valid_square && right_en_passant_able && net_y = 1
+          && net_x = 1
+          && board.(curr_x + 1).(curr_y + 1) = None
+        then true
+        else false (* Check when color is Black *)
+      else if
+        (* Go forward two *)
+        basic_valid_square && curr_y = 6 && net_y = -2 && net_x = 0
+        && board.(curr_x).(curr_y - 1) = None
+        && board.(curr_x).(curr_y - 2) = None
+      then true (* Go forward one *)
+      else if
+        basic_valid_square && net_y = -1 && net_x = 0
+        && board.(curr_x).(curr_y - 1) = None
+      then true (* Diagonal down-left -- no en passant *)
+      else if
+        basic_valid_square
+        && (not left_en_passant_able)
+        && net_y = -1 && net_x = -1
+        && board.(curr_x - 1).(curr_y - 1) != None
+      then true (* Diagonal down-left -- Yes en passant *)
+      else if
+        basic_valid_square && left_en_passant_able && net_y = -1
+        && net_x = -1
+        && board.(curr_x - 1).(curr_y - 1) = None
+      then true (* Diagonal down-right -- no en passant *)
+      else if
+        basic_valid_square
+        && (not right_en_passant_able)
+        && net_y = -1 && net_x = 1
+        && board.(curr_x + 1).(curr_y - 1) != None
+      then true (* Diagonal down-right -- Yes en passant *)
+      else if
+        basic_valid_square && right_en_passant_able && net_y = -1
+        && net_x = 1
+        && board.(curr_x + 1).(curr_y - 1) = None
+      then true
+      else false
+    in
+
+    check_conditions
+
+  (* =================================================== *)
+  let potential_squares
+      (curr_x, curr_y)
+      board_arr
+      (color : color)
+      last_move : (int * int) list =
+    (* White Pawn potential moves *)
+    if color = White then
+      List.filter
+        (is_valid_square_pawn (curr_x, curr_y) board_arr color last_move)
+        [
+          (curr_x, curr_y + 2);
+          (curr_x, curr_y + 1);
+          (curr_x + 1, curr_y + 1);
+          (curr_x - 1, curr_y + 1);
+        ] (* Black Pawn potential moves *)
+    else
+      List.filter
+        (is_valid_square_pawn (curr_x, curr_y) board_arr color last_move)
+        [
+          (curr_x, curr_y + 2);
+          (curr_x, curr_y + 1);
+          (curr_x + 1, curr_y + 1);
+          (curr_x - 1, curr_y + 1);
```

Comment:
These are the same as the white pawn moves. I think this was a copy-pasting mistake.